src/cattrs/converters.py

```diff
@@ -631,8 +631,8 @@ def unstructure_attrs_astuple(self, obj: Any) -> tuple[Any, ...]:
         return tuple(res)
 
     def _unstructure_enum(self, obj: Enum) -> Any:
-        """Convert an enum to its value."""
-        return obj.value
+        """Convert an enum to its unstructured value."""
+        return self._unstructure_func.dispatch(obj.value.__class__)(obj.value)
```

Comment:
Hm, I'm a little concerned about performance and backwards comp. Can we do a deep unstructure only if the enum has a `_value_` attribute, like when structuring?

Reply:
Same as above. Change made so it's conditional like the above as well.